### DISCUSSION TITLE
[store] Use 'any' instead of generics

### DISCUSSION THREAD

**Discussion by TheAlmightyCrumb** [DefinitelyTyped/DefinitelyTyped#59172] [Issues with a @types package]
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/10ea84a4eb0b949af739c5837d5ade42438bc4ae/types/store/v1/index.d.ts#L12-L13

As seen above, the `set` and `get` functions both use `any` instead of generics.
For a better, safer use consider the following:
```
set<T>(key: string, value: T): T;
get<T>(key: string, defaultVal?: T): T | null;
```
In addition, other properties of this interface use `any`. Consider changing them as well.

[a comment by an automated account is not shown]

**Comment by harry0000** [comment]
What store.js does is a kind of side effect. So it is safer to type casting or narrowing the `get` / `set` value from store.js at the user's own risk.

When generics is used, the code is no longer safe because it is synonymous with the following code.

```ts
const value: T = store.get('foo') as T;
```

A more safer code is as follows.

```ts
function isExpectedType(arg: any): arg is ExpectedType {
  // check arg is expected type or not.
}

const anyValue = store.get('foo');

if (isExpectedType(anyValue)) {
  // do something.
}
```
